### ISSUE TITLE
Tons of incorrect typecheck errors inside miri

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust-analyzer#22812]
**rust-analyzer version**: `rust-analyzer version: 0.4.2973-standalone [/home/gh-bjorn3/.vscode-server/extensions/rust-lang.rust-analyzer-0.4.2973-linux-arm64/server/rust-analyzer]`

**rustc version**: `rustc 1.98.0-beta.2 (08d5b675a 2026-07-12)`

**editor or extension**: VSCode

**relevant settings**: Fairly standard rust-lang/rust config

**repository link (if public, optional)**: https://github.com/rust-lang/rust

If I open for example `src/tools/miri/src/machine.rs`, I'm getting several errors like `expected &[OpTy<'_, CtfeProvenance>], found &Vec<OpTy<'_, Provenance>, Global>` from rust-analyzer. Other files are way worse.

**Comment by ShoyuVanilla** [comment]
Hmm, maybe we are having trouble locating lang items in Miri, such as `CoerceUnsized` trait, which is necessary for unsized coerciosn 🤔

**Comment by Veykril** [comment]
Yea that sounds like we are missing the `Deref` impl?

**Comment by ChayimFriedman2** [comment]
Does it work with stable rustc?

**Comment by bjorn3** [comment]
Miri doesn't work with stable rustc. The beta I mentioned is the bootstrap compiler that current main of rustc uses.

**Comment by ChayimFriedman2** [comment]
And if you point r-a at stable std (`"rust-analyzer.server.extraEnv": { "RUSTUP_TOOLCHAIN": "stable" }`)?

**Comment by bjorn3** [comment]
```jsonc
"rust-analyzer.server.extraEnv": {
        //"RUSTC": "${workspaceFolder}/build/host/stage0/bin/rustc",
        //"CARGO": "${workspaceFolder}/build/host/stage0/bin/cargo",
        "RUSTUP_TOOLCHAIN": "stable"
    },
```

has the same problem. I think the problem is `Provenance` vs `CtfeProvenance` confusion. I'm also seeing errors like `expected &MPlaceTy<'_, CtfeProvenance>, found &MPlaceTy<'tcx, Provenance>`.

**Comment by ChayimFriedman2** [comment]
That's weird because `CtfeProvnenace` is barely used. Does it work in the Miri repo?

**Comment by Veykril** [comment]
Miri tends to use a ton of glob imports which r-a struggles with, so those kinds of type confusions aren't new.

**Comment by bjorn3** [comment]
It seems rust-analyzer thinks `Pointer` directly refers directly to `rustc_middle::mir::interpret::pointer::Pointer<Prov>` which has a default of `Prov=CtfeProvenance` rather than refering to the `miri::Pointer` type alias which uses `Prov=Option<miri::machine::Provenance>` instead.
